Oficial fixes to make Gnuplot work with wxWidgets 3.0.
Can be removed with Gnuplot update to version 5.0 or later.

https://github.com/gnuplot/gnuplot/commit/a275a85b335fb358cea09d7e64204caca3193e1a
https://github.com/gnuplot/gnuplot/commit/632f86b64659b9ad14bfae1c850ce268f0d39352

--- gnuplot-4.6.0/src/wxterminal/wxt_gui.h
+++ gnuplot-4.6.0/src/wxterminal/wxt_gui.h
@@ -203,6 +203,8 @@
  * ====================================================================*/
 
 #ifdef WXT_MULTITHREADED
+#include <X11/Xlib.h>	/* Magic fix for linking against wxgtk3.0 */
+
 /* thread class, where the gui loop runs.
  * Not needed with Windows, where the main loop
  * already processes the gui messages */
@@ -234,6 +236,10 @@
 class wxtApp : public wxApp
 {
 public:
+#ifdef WXT_MULTITHREADED 
+	/* Magic fix needed by wxgtk3.0 */
+	wxtApp() : wxApp() { XInitThreads(); } 
+#endif 
 	/* This one is called just after wxWidgets initialization */
 	bool OnInit();
 	/* cleanup on exit */
--- gnuplot-4.6.0/src/wxterminal/wxt_gui.cpp
+++ gnuplot-4.6.0/src/wxterminal/wxt_gui.cpp
@@ -2493,6 +2498,20 @@
 /* refresh the plot by (re)processing the plot commands list */
 void wxtPanel::wxt_cairo_refresh()
 {
+	/* This check may prevent the assert+die behavior seen with wxgtk3.0 */
+	/* Symptom:
+	  ./src/gtk/dcclient.cpp(2043): assert "m_window" failed in DoGetSize(): GetSize() doesn't work without window [in thread 7fb21f386700]
+	  Call stack:
+	  [00] wxOnAssert(char const*, int, char const*, char const*, wchar_t const*)
+	  [01] wxClientDCImpl::DoGetSize(int*, int*) const
+	  [02] wxBufferedDC::UnMask()                  
+	  wxwidgets documentation to the contrary, panel->IsShownOnScreen() is unreliable
+	 */
+	if (!wxt_current_window) {
+		FPRINTF((stderr,"wxt_cairo_refresh called before window exists\n"));
+		return;
+	}
+
 	/* Clear background. */
 	gp_cairo_solid_background(&plot);
 
